src/core/game.cc

```diff
@@ -869,7 +869,7 @@ struct BatchExecutor {
                           result_[dstp]);
 #endif
             } else {
-              if ((int)i->pi[slot].size() < seqlen * 16 + 1 ||
```

Comment:
very confused what was going on here – how do you follow an int?

Reply:
The variable `i` is a `vector<...>::iterator`, which is basically a pointer to an array element. It's declared in the for loop as an `auto` not an `int`.

What's getting converted to `(int)` is the whole expression `i->pi[slot].size()`. It's one of the reasons I don't like this syntax and prefer to use `std::static_cast<int>(expression)`.

Reply:
wooow I see. And by default it would be a size_t I guess, which can be automatically cast to an int but for some reason they were not?

Reply:
I don't know why they were cast to an int because `seqlen` is an `uint64_t`, so the compiler was raising a warning.

Reply:
pffft